src/map/mob.c

```diff
@@ -3646,77 +3646,354 @@ void item_dropratio_adjust(int nameid, int mob_id, int *rate_adjust)
 
-	exp = (double)atoi(str[8]) * (double)battle_config.job_exp_rate / 100.;
-	db->job_exp = (unsigned int)cap_value(exp, 0, UINT_MAX);
+	if (mob->lookup_const(mobt, "Hp", &i32) && i32 >= 0) {
+		mstatus->max_hp = i32;
+	} else {
+		mstatus->max_hp = 1;
+	}
 
-	mstatus->rhw.range = atoi(str[9]);
+	if (mob->lookup_const(mobt, "Sp", &i32) && i32 >= 0) {
+		mstatus->max_sp = i32;
+	}
 
-	mstatus->rhw.atk = mob_parse_dbrow_cap_value(class_,UINT16_MIN,UINT16_MAX,atoi(str[10]));
-	mstatus->rhw.atk2 = mob_parse_dbrow_cap_value(class_,UINT16_MIN,UINT16_MAX,atoi(str[11]));
+	if (mob->lookup_const(mobt, "Exp", &i32) && i32 >= 0) {
+		exp = (double)(i32) * (double)battle_config.base_exp_rate / 100.;
+		entry->base_exp = (unsigned int)cap_value(exp, 0, UINT_MAX);
+	}
 
-	mstatus->def  = mob_parse_dbrow_cap_value(class_,DEFTYPE_MIN,DEFTYPE_MAX,atoi(str[12]));
-	mstatus->mdef = mob_parse_dbrow_cap_value(class_,DEFTYPE_MIN,DEFTYPE_MAX,atoi(str[13]));
+	if (mob->lookup_const(mobt, "JExp", &i32) && i32 >= 0) {
+		exp = (double)(i32) * (double)battle_config.job_exp_rate / 100.;
+		entry->job_exp = (unsigned int)cap_value(exp, 0, UINT_MAX);
+	}
 
-	mstatus->str  = mob_parse_dbrow_cap_value(class_,UINT16_MIN,UINT16_MAX,atoi(str[14]));
-	mstatus->agi  = mob_parse_dbrow_cap_value(class_,UINT16_MIN,UINT16_MAX,atoi(str[15]));
-	mstatus->vit  = mob_parse_dbrow_cap_value(class_,UINT16_MIN,UINT16_MAX,atoi(str[16]));
-	mstatus->int_ = mob_parse_dbrow_cap_value(class_,UINT16_MIN,UINT16_MAX,atoi(str[17]));
-	mstatus->dex  = mob_parse_dbrow_cap_value(class_,UINT16_MIN,UINT16_MAX,atoi(str[18]));
-	mstatus->luk  = mob_parse_dbrow_cap_value(class_,UINT16_MIN,UINT16_MAX,atoi(str[19]));
+	if (mob->lookup_const(mobt, "AttackRange", &i32) && i32 >= 0) {
+		mstatus->rhw.range = i32;
+	} else {
+		mstatus->rhw.range = 1;
+	}
+
+	if ((t = libconfig->setting_get_member(mobt, "Attack"))) {
+		if (config_setting_is_aggregate(t)) {
+			if (libconfig->setting_length(t) >= 2)
+				mstatus->rhw.atk2 = libconfig->setting_get_int_elem(t, 1);
+			if (libconfig->setting_length(t) >= 1)
+				mstatus->rhw.atk = libconfig->setting_get_int_elem(t, 0);
+		} else if (mob->lookup_const(mobt, "Attack", &i32) && i32 >= 0) {
+			mstatus->rhw.atk = i32;
+			mstatus->rhw.atk2 = i32;
+		}
+	}
+
+	if (mob->lookup_const(mobt, "Def", &i32) && i32 >= 0) {
+		mstatus->def = mob_parse_dbrow_cap_value(class_, DEFTYPE_MIN, DEFTYPE_MAX, i32);
+	}
+	if (mob->lookup_const(mobt, "Mdef", &i32) && i32 >= 0) {
+		mstatus->mdef = mob_parse_dbrow_cap_value(class_, DEFTYPE_MIN, DEFTYPE_MAX, i32);
+	}
+
+	if ((t = libconfig->setting_get_member(mobt, "Stats"))) {
+		if (config_setting_is_group(t)) {
+			mob->read_db_stats_sub(entry, mstatus, class_, t);
+		} else if (mob->lookup_const(mobt, "Stats", &i32) && i32 >= 0) {
+			mstatus->str = mstatus->agi = mstatus->vit = mstatus->int_ = mstatus->dex = mstatus->luk =
+				mob_parse_dbrow_cap_value(class_, UINT16_MIN, UINT16_MAX, i32);
+		}
+	}
```

Comment:
Original stat-parsing code was ensuring these values wouldn't go below 1, to avoid divisions by zero and such (at least in pre-renewal builds). If we remove that, we should investigate to make sure it's safe to.